src/schema/posts.ts

```diff
@@ -1371,6 +1391,60 @@ export const resolvers: IResolvers<any, Context> = {
 
       return { _: true };
     },
+    swapPinnedPosts: async (
+      _,
+      { id, swapWithId }: SwapPinnedPostArgs,
+      ctx: Context,
+    ): Promise<GQLEmptyResponse> => {
+      await ctx.con.transaction(async (manager) => {
+        const repo = manager.getRepository(Post);
+        const post = await repo.findOneBy({ id });
+        const swapWithPost = await repo.findOneBy({
+          id: swapWithId,
+        });
+
+        await ensureSourcePermissions(
+          ctx,
+          post.sourceId,
+          SourcePermissions.PostPin,
+        );
+
+        if (!post.pinnedAt || !swapWithPost.pinnedAt) {
+          throw new ValidationError('Posts must be pinned first');
+        }
+
+        const isNextPost = swapWithPost.pinnedAt > post.pinnedAt;
+        const swapPinnedTime = new Date(
+          swapWithPost.pinnedAt.getTime() + 1000 * (isNextPost ? 1 : -1),
+        );
+        const operation = isNextPost
+          ? `+ interval '1 second'`
+          : `- interval '1 second'`;
+        const comparison = isNextPost ? '>=' : '<=';
+
+        await manager
+          .createQueryBuilder()
+          .update(Post)
+          .set({
+            pinnedAt: () => `"pinnedAt" ${operation}`,
+          })
```

Comment:
You can now use if else and conditionally call `set` with different `operation` on query builder to avoid using template string in query.